pkg/kubelet/cadvisor/cadvisor_linux.go

```diff
@@ -117,6 +124,33 @@ func New(port uint, runtime string) (Interface, error) {
 	return cadvisorClient, nil
 }
 
+func generateCollectorHttpClient(cAdvisorCollectorClientCertFile, cAdvisorCollectorClientPrivateKeyFile string) (*http.Client, error) {
+	//Enable accessing insecure endpoints. We should be able to access metrics from any endpoint
+	tlsConfig := &tls.Config{
+		InsecureSkipVerify: true,
```

Comment:
The argument in the linked comment seems to be:
 1. this isn't worse than http
 1. getting valid serving certs is hard

While both those points are true, it doesn't seem like a sufficient reason to hardcode something that can never verify the identity of the server its communicating with.  I don't mind having this an option, but allowing someone to plumb in a CA seems pretty reasonable.

For instance, openshift can mint serving certs for services.  You could use a service DNS name and the CA we sign with and you'll be able to verify you're speaking to the service you think you are.